icu4j/main/core/src/main/java/com/ibm/icu/util/PersianCalendar.java

```diff
@@ -93,6 +95,22 @@ public class PersianCalendar extends Calendar {
     
     private static final int PERSIAN_EPOCH = 1948320;
 
+    private static final int NON_LEAP_YEARS[] = {
+       1502, 1601, 1634, 1667, 1700, 1733, 1766, 1799, 1832, 1865, 1898, 1931, 1964, 1997, 2030, 2059,
+       2063, 2096, 2129, 2158, 2162, 2191, 2195, 2224, 2228, 2257, 2261, 2290, 2294, 2323, 2327, 2356,
+       2360, 2389, 2393, 2422, 2426, 2455, 2459, 2488, 2492, 2521, 2525, 2554, 2558, 2587, 2591, 2620,
+       2624, 2653, 2657, 2686, 2690, 2719, 2723, 2748, 2752, 2756, 2781, 2785, 2789, 2818, 2822, 2847,
+       2851, 2855, 2880, 2884, 2888, 2913, 2917, 2921, 2946, 2950, 2954, 2979, 2983, 2987,
+    };
+
+    private static Set<Integer> LEAP_CORRECTION = null;
+    {
+        Set<Integer> prefab = new HashSet<Integer>(NON_LEAP_YEARS.length);
+        for (int nonLeap : NON_LEAP_YEARS) {
+            prefab.add(nonLeap);
+        }
```

Comment:
I think this means that every time you do a lookup, you end up auto-boxing the year to an Integer object. And I think these are large enough not to be cached.

How about using a [java.util.BitSet](https://docs.oracle.com/javase/8/docs/api/java/util/BitSet.html)?
You could minimize it by also keeping the min value, and storing and looking up (year-minYear).

Something like
```
private static final class NonLeapYears {
    int minYear = NON_LEAP_YEARS[0];
    int maxYear = NON_LEAP_YEARS[NON_LEAP_YEARS.length - 1];
    BitSet offsetYears;

    NonLeapYears() {
        offsetYears = new BitSet(maxYear - minYear + 1);
        for (int nonLeap : NON_LEAP_YEARS) {
            offsetYears.set(nonLeap - minYear);
        }
    }

    boolean contains(int year) {
        return minYear <= year && year <= maxYear && offsetYears.get(year - minYear);
    }
}
```
and then here you initialize LEAP_CORRECTION = new NonLeapYears();
and below you just call LEAP_CORRECTION.contains(year)
without an additional year range check

WDYT?